modules/related-posts/jetpack-related-posts.php

```diff
@@ -246,6 +251,118 @@ public function get_target_html_unsupported() {
 		return "\n\n<!-- Jetpack Related Posts is not supported in this context. -->\n\n";
 	}
 
+	/**
+	 * ===============
+	 * GUTENBERG BLOCK
+	 * ===============
+	 */
+
+	/**
+	 * Register the Related Posts Gutenberg block on server.
+	 *
+	 * @action init
+	 * @uses register_block_type
+	 * @return null
+	 */
+	public function register_block() {
+		register_block_type(
+			'a8c/related-posts',
+			array(
+				'render_callback' => array( $this, 'render_block' ),
+			)
+		);
+	}
+
+	/**
+	 * Render the related posts markup.
+	 *
+	 * @return string
+	 */
+	public function render_block( $attributes ) {
+		$block_attributes = array(
+			'show_thumbnails' => isset( $attributes['displayThumbnails'] ) ? (bool) $attributes['displayThumbnails'] : false,
+			'show_date' => isset( $attributes['displayDate'] ) ? (bool) $attributes['displayDate'] : true,
+			'show_context' => isset( $attributes['displayContext'] ) ? (bool) $attributes['displayContext'] : true,
+			'layout' => isset( $attributes['postLayout'] ) && $attributes['postLayout'] === 'list' ? $attributes['postLayout'] : 'grid',
+			'size' => ! empty( $attributes['postsToShow'] ) ? absint( $attributes['postsToShow'] ) : 3,
+		);
+
+		$related_posts = $this->get_for_post_id( get_the_ID(), array(
+			'size' => $block_attributes['size'],
+		) );
+
+		if ( ! $related_posts ) {
+			return '';
+		}
+
+		ob_start();
+		?>
+		<div id="jp-relatedposts" class="jp-relatedposts" style="display: block;">
+			<div class="jp-relatedposts-items jp-relatedposts-items-visual jp-relatedposts-<?php echo esc_attr( $block_attributes['layout'] ); ?>">
+				<?php foreach ( $related_posts as $index => $related_post ): 
+					$classes = array_filter( array(
+						'jp-relatedposts-post',
+						'jp-relatedposts-post' . $index,
+						! empty( $block_attributes['show_thumbnails'] ) ? 'jp-relatedposts-post-thumbs' : '',
+					) );
+					?>
```

Comment:
After https://github.com/Automattic/jetpack/pull/10132/commits/73b1f29b3c01fa2845b75b14070eaf264cbbabeb this will not happen anymore, because we no longer generate any HTML with JS.